packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -162,14 +162,59 @@ export class TransformerResolver implements TransformerResolverProvider {
       slotEntry = [];
     }
 
-    slotEntry.push({
-      requestMappingTemplate,
-      responseMappingTemplate,
-      dataSource,
-    });
+    if (!this.updateSlotIfExists(slotName, requestMappingTemplate, responseMappingTemplate)) {
+      slotEntry.push({
+        requestMappingTemplate,
+        responseMappingTemplate,
+        dataSource,
+      });
+    }
     this.slotMap.set(slotName, slotEntry);
   };
 
+  updateSlotIfExists = (
+    slotName: string,
+    requestMappingTemplate?: MappingTemplateProvider,
+    responseMappingTemplate?: MappingTemplateProvider,
+  ): boolean => {
+    const slotEntires = this.slotMap.get(slotName)!;
+    let slotIndex = 1;
+
+    if (!slotEntires) {
+      return false;
+    }
+
+    for (const slotItem of slotEntires) {
```

Comment:
Rather than needing a mutable `slotIndex` can we just call `slotEntries.forEach((slotEntry, slotIndex) => { /** your logic below **/ };`

Reply:
This block requires early termination if match is found. The only way to early exit foreach is using exception, that would look ugly. I have also considered the usage of every() and some(), however `for .. of ...` seems to be the best fit here. I'm open to suggestions if there is a better way.

Reply:
Ahh, I see. I'm fine w/ this as-is in that case.